### PR TITLE
AJ-1003 Standardize File Upload Banner for GCP and Azure Workspaces

### PR DESCRIPTION
**Pull request by kevinmarete**
### What
- In GCP workspaces, the upload banner currently appears only for files larger than 512k, while in Azure, it shows for all uploads. This PR updates GCP to display the upload banner for all file sizes, matching Azure's behavior.

### Why
- A user requested the banner to always appear for file uploads to confirm success without checking tables. We want to ensure consistent upload confirmation across GCP and Azure workspaces.

### Testing strategy
- Manual testing of functionality by uploading TSV files in GCP workspaces.

https://github.com/user-attachments/assets/e19039b5-78ef-4277-b97b-2a5c25c876c7

### CODE REVIEW

File: src/workspace-data/Data.js [hunk 1]
```diff
@@ -1170,10 +1170,13 @@ export const WorkspaceData = _.flow(
                   uploadingFile &&
                     h(EntityUploader, {
                       onDismiss: () => setUploadingFile(false),
-                      onSuccess: () => {
+                      onSuccess: (recordType) => {
                         setUploadingFile(false);
                         forceRefresh();
                         loadMetadata();
+                        notify('success', `Data imported successfully to table ${recordType}.`, {
```
Comment (calypsomatic) [inline src/workspace-data/Data.js]: This always shows the message immediately - even for an async upload that takes a while.  For async uploads, it shouldn't show a success message until it's actually complete.  Async uploads already show a success message upon completion, so with this change we get two success messages

Comment (kevinmarete) [reply src/workspace-data/Data.js]: Added check for synchronous uploads: https://github.com/DataBiosphere/terra-ui/pull/5124/commits/77101dc80ce12a4d7e015306274da6550c4f1721

**Review by davidangb — COMMENTED** [review]
For Azure/WDS workspaces, this results in double/redundant success notifications. You could potentially resolve that by checking `isGoogleWorkspace` or by introspecting the `dataProvider`.

We are not prioritizing Azure workspaces, so if resolving the double notifications is onerous, I am ok shipping this as-is.

**Review by davidangb — APPROVED** [review]
thank you!

**Review by calypsomatic — APPROVED** [review]

**Pull request merged**